pkg/runtime/hotreload/reconciler/reconciler.go

```diff
@@ -140,34 +140,42 @@ func (r *Reconciler[T]) reconcile(ctx context.Context, result *differ.Result[T])
 		{result.Updated, operatorpb.ResourceEventType_UPDATED},
 		{result.Created, operatorpb.ResourceEventType_CREATED},
 	} {
-		wg.Add(len(group.resources))
+		errCh := make(chan error)
 		for _, resource := range group.resources {
 			go func(resource T, eventType operatorpb.ResourceEventType) {
-				defer wg.Done()
-				r.handleEvent(ctx, &loader.Event[T]{
+				errCh <- r.handleEvent(ctx, &loader.Event[T]{
 					Type:     eventType,
 					Resource: resource,
 				})
 			}(resource, group.eventType)
 		}
-		wg.Wait()
+
+		errs := make([]error, 0, len(group.resources))
+		for range group.resources {
```

Comment:
```suggestion
		for i := 0; i < len(group.resources); i++ {
```

Reply:
What is the benefit of creating the variable `i` here?